app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -11,12 +11,23 @@ def connect(options = {})
 
   def verify_credentials(_auth_type = nil, options = {})
     require 'azure-armrest'
-    connect(options)
-  rescue Azure::Armrest::UnauthorizedException
-    raise MiqException::MiqHostError, _("Incorrect credentials - check your Azure Client ID and Client Key")
+    conf = connect(options)
+
+    unless subscription.blank?
+      vms = ::Azure::Armrest::VirtualMachineService.new(conf)
+      subscriptions = vms.subscriptions.map(&:subscription_id)
+      unless subscriptions.include?(subscription)
+        raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - check your Azure Subscription ID")
+      end
+    end
+  rescue Azure::Armrest::UnauthorizedException, Azure::Armrest::BadRequestException
```

Comment:
I think Bill's question is, "does it *always* indicate invalid credentials?"

But, really, this looks like a bug in Azure.  HTTP 400 means "bad syntax for the request", which is hugely vague.  But, since there are specific HTTP responses reserved for invalid login credentials, this seems completely wrong that they'd use HTTP 400 for invalid login credentials.

Can you check if there's anything in the HTTP response that indicates that this particular 400 response is really for bad login?  If there's no other info in the response, then I guess you have to handle it this way.